apollo-portal/src/main/resources/static/scripts/controller/UserController.js

```diff
@@ -31,6 +31,35 @@ function UserController($scope, $window, $translate, toastr, AppUtil, UserServic
         })
     }
 
+    var num_chars = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+    var alpha_chars =[ 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l',
+               'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'];
+
+    function randomAlphanumeric(len) {
```

Comment:
I'm not sure whether we should provide this functionality, because most modern browsers (like Safari) or password manager (like 1password) has this ability to generate strong random password, this function defined here is rather naive IMO

Reply:
I agree with you if it's self-registration, but in this scenario, it is the administrator manage the account. This is just a convenience for administrators to generate default passwords for members if he wanted. And the browser's password generation extension may only work for the password component. If you don't want it, I'll remove it. Thanks.

Reply:
@WillardHu  Let's wait for other reviewers' opinions

Reply:
I would remove that functionality since `strong password` generation can be done by the browser. 
Additionally a custom password generator would be harder to maintain (for example if the password policy changes etc) with little to none benefit.

Reply:
I would also suggest we don't provide this functionality as it is not a core function of apollo and there are many plugins or browsers doing this job much better. What we need to do though is to make sure the password form is compatible with those plugins or browsers.

Reply:
Ok, the frontend modifications have been rollback.